src/components/stepTwo/index.js

```diff
@@ -1,142 +1,78 @@
-    this.updateDecimalsStore({
-      value: this.state.decimals,
-      pristine: false,
-      valid: this.state.validation.decimals.valid
-    })
-  }
-
-  updateTokenStore = (event, property) => {
-    const value = event.target.value;
-    this.props.tokenStore.setProperty(property, value);
-    this.props.tokenStore.validateTokens(property);
+    checkWeb3(this.props.web3Store.web3)
   }
 
-  updateDecimalsStore = ({ value, pristine, valid }) => {
-    const newState = update(this.state, {
-      validation: {
-        decimals: {
-          $set: {
-            disabled: this.state.validation.decimals.disabled,
-            pristine: pristine,
-            valid: valid
-          },
-        },
-      },
-    })
-    newState.decimals = value
-
-    this.setState(newState)
+  beforeNavigate = (e) => {
+    e.preventDefault()
+    e.stopPropagation()
 
-    // TODO: store should only be updated when the user hits 'Continue'
-    const { tokenStore } = this.props
-    tokenStore.setProperty('decimals', value)
-    tokenStore.updateValidity('decimals', valid)
-  }
-
-  disableDecimals = () => {
-    if (this.state.decimals === '') {
-      this.updateDecimalsStore({ value: 0, pristine: false, valid: VALID })
-    }
-
-    this.setState(update(this.state, { validation: { decimals: { disabled: { $set: true } } } }))
-  }
-
-  enableDecimals = () => {
-    this.setState(update(this.state, { validation: { decimals: { disabled: { $set: false } } } }))
-  }
-
-  renderLink () {
-    return <Link className="button button_fill" to='/3'>Continue</Link>
-  }
-
-  renderLinkComponent = () => {
-    if(this.props.tokenStore.isTokenValid) {
-      return this.renderLink();
-    }
-    return <div onClick={this.showErrorMessages.bind(this, 'token')} className="button button_fill"> Continue</div>
+    document
+      .getElementById('tokenData')
+      .dispatchEvent(new Event('submit'))
   }
 
   removeReservedToken = index => {
-    const lastReservedItem = this.props.reservedTokenStore.tokens.length === 1
-
-    if (lastReservedItem) {
-      this.enableDecimals()
-    }
-
     this.props.reservedTokenStore.removeToken(index)
   }
 
   clearReservedTokens = () => {
     return clearingReservedTokens()
       .then(result => {
         if (result.value) {
-          this.enableDecimals()
-
           this.props.reservedTokenStore.clearAll()
         }
       })
   }
 
   addReservedTokensItem = newToken => {
-    const firstReservedItem = !this.props.reservedTokenStore.tokens.length
+    this.props.reservedTokenStore.addToken(newToken)
+  }
 
-    if (firstReservedItem) {
-      this.disableDecimals()
-    }
+  updateTokenStore = (property, value) => {
+    const validity = validators(property, value) ? VALID : INVALID
 
-    this.props.reservedTokenStore.addToken(newToken)
+    this.props.tokenStore.setProperty(property, value)
+    this.props.tokenStore.updateValidity(property, validity)
+  }
+
+  onSubmit = () => {
```

Comment:
This isn't doing anything right now, is it?

Reply:
`onSubmit`? when the `submit` event is [dispatched](https://github.com/poanetwork/token-wizard/pull/719/files/a4970602f816d6a477070fcc160b02c0c8a8bf73#diff-c0618d3a6e868cb93320cafc2d83a613R42), if all the the fields in the form are valid, this method [will be called](https://github.com/poanetwork/token-wizard/pull/719/files/a4970602f816d6a477070fcc160b02c0c8a8bf73#diff-c0618d3a6e868cb93320cafc2d83a613R90)